libs/core/threading_base/include/hpx/threading_base/annotated_function.hpp

```diff
@@ -33,11 +33,11 @@ namespace hpx { namespace util {
 #if defined(HPX_HAVE_THREAD_DESCRIPTION)
     ///////////////////////////////////////////////////////////////////////////
 #if defined(HPX_COMPUTE_DEVICE_CODE)
-    struct annotate_function
+    struct HPX_NODISCARD annotate_function
     {
         HPX_NON_COPYABLE(annotate_function);
 
-        explicit annotate_function(char const*) {}
+        explicit annotate_function(std::string) {}
```

Comment:
Shouldn't this be left as `char const*` to avoid allocations?

Reply:
Yes, and the change is an inconsistency in the first place. For `annotate_function` one can anyway use `std::string::c_str` since `annotate_function` is meant for the scope where it's defined.